spot_rl_experiments/spot_rl/skills/atomic_skills.py

```diff
@@ -279,60 +303,58 @@ def reset_env_and_policy(self, target_obj_name):
 
-    def execute_pick(self, target_object_list, take_user_input=False):
+    def execute_rl_loop(self, target_object: str, take_user_input=False):
         """
-        Gaze at the target object list and pick up the objects if specified in the config
+        !!BLOCKING CALL!!
 
-        CAUTION: The robot will drop the object after picking it, please use objects that are not fragile
+        Executes the action based on output from policy
 
         Args:
-            target_object_list (list): List of target objects to gaze at
+            target_object (str): target object to gaze at and pick
             take_user_input (bool): Whether to take user input for the success of the gaze
+        """
 
-        Returns:
-            gaze_success_list (list): List of dictionaries containing the target object name, time taken and success
-        """
-        gaze_success_list = []
-        print(f"Target object list : {target_object_list}")
-        for target_object in target_object_list:
-            observations = self.reset_env_and_policy(target_obj_name=target_object)
-            done = False
-            start_time = time.time()
-            self.env.say(f"Gaze at target object - {target_object}")
-
-            while not done:
-                action = self.policy.act(observations)
-                if self._use_mobile_pick:
-                    arm_action, base_action = None, None
-                    # first 4 are arm actions, then 2 are base actions & last bit is unused
-                    arm_action = action[0:4]
-                    base_action = action[4:6]
-
-                    observations, _, done, _ = self.env.step(
-                        arm_action=arm_action, base_action=base_action
-                    )
-                else:
-                    observations, _, done, _ = self.env.step(arm_action=action)
-            self.env.say("Gaze finished")
-            # Ask user for feedback about the success of the gaze and update the "success" flag accordingly
-            success_status_from_user_feedback = True
-            if take_user_input:
-                user_prompt = f"Did the robot successfully pick the right object - {target_object}?"
-                success_status_from_user_feedback = map_user_input_to_boolean(
-                    user_prompt
+        observations = self.reset_env_and_policy(target_obj_name=target_object)
+        done = False
+
+        # Update logged data at init
+        self.do_stuff_pre_loop(target_object)
+
+        while not done:
+            action = self.policy.act(observations)
+            if self._use_mobile_pick:
+                arm_action, base_action = None, None
+                # first 4 are arm actions, then 2 are base actions & last bit is unused
+                arm_action = action[0:4]
+                base_action = action[4:6]
+
+                observations, _, done, _ = self.env.step(
+                    arm_action=arm_action, base_action=base_action
                 )
```

Comment:
Per discussion with @jimmytyyang 
1. Update env.step() to take in dictionary
2. Update each child skill (atomic) to implement `split_actions` .. and split actions should output this dictionary.

Reply:
Added this to PR scope in PR description.

Reply:
Addressed